flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```
@@ -493,6 +497,59 @@ public boolean accept(File dir, String name) {
 		}
 	}
 
+	public static boolean verifyTokenKindInContainerCredentials(final String[] tokens, final String containerId) {
+		List<String> tokenList = Arrays.asList(tokens);
+		File cwd = new File("target/" + YARN_CONFIGURATION.get(TEST_CLUSTER_NAME_KEY));
+		if (!cwd.exists() || !cwd.isDirectory()) {
+			return false;
+		}
+
+		File containerTokens = findFile(cwd.getAbsolutePath(), new FilenameFilter() {
+			@Override
+			public boolean accept(File dir, String name) {
+				return name.equals(containerId + ".tokens");
+			}
+		});
+
+		if (containerTokens != null) {
+			LOG.info("Verifying tokens in {}", containerTokens.getAbsolutePath());
+
+			Credentials tmCredentials = null;
+			try {
+				tmCredentials = Credentials.readTokenStorageFile(containerTokens, new Configuration());
+			} catch (IOException e) {
+				LOG.warn("Unable to read credential file: " + e.getMessage() + " file: " + containerTokens.getAbsolutePath());
```

SHouldn't we simply let the exception bubble up and let the test fail?